plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -234,6 +235,11 @@
                 storageClass);
     }
 
+    private static Map<String, String> buildLocation(RepositoryMetadata metadata) {
+        return Map.of("base_path", buildBasePath(metadata).buildAsString(),
```

Comment:
Should we just expose the setting from the user? I'm not sure if we want to do some custom sanitation using `buildBasePath` here.

Reply:
is there any harm on sanitizing the user setting? it deals with empty and null settings

Reply:
I would expect an external system to use this to match whatever configuration it set on the cluster (in order to identify the repo). Adapting this in any way feels trappy to me. On a related note (but not relevant for this PR), we should look at these `buildBasePath` methods (which are all slightly different based on repo type, and a source of trappiness with these odd rules that they implement).